src/OpenFeature.Contrib.Providers.Flagd/Cache.cs

```diff
@@ -0,0 +1,170 @@
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("DynamicProxyGenAssembly2")]
+
+namespace OpenFeature.Contrib.Providers.Flagd
+{
+    internal interface ICache<TKey, TValue>
+    {
+        void Add(TKey key, TValue value);
+        TValue TryGet(TKey key);
+        void Delete(TKey key);
+        void Purge();
+    }
+    class LRUCache<TKey, TValue> : ICache<TKey, TValue> where TValue : class
+    {
+        private readonly int _capacity;
+        private readonly Dictionary<TKey, Node> _map;
+        private Node _head;
+        private Node _tail;
+
+        private System.Threading.Mutex _mtx;
+
+        public LRUCache(int capacity)
+        {
+            _capacity = capacity;
+            _map = new Dictionary<TKey, Node>();
+            _mtx = new System.Threading.Mutex();
+        }
+
+        public TValue TryGet(TKey key)
+        {
+            try
+            {
+                _mtx.WaitOne();
+                if (_map.TryGetValue(key, out Node node))
+                {
+                    MoveToFront(node);
+                    return node.Value;
+                }
+                return default(TValue);
+            }
+            finally
+            {
+                _mtx.ReleaseMutex();
+            }
+
+        }
+
+        public void Add(TKey key, TValue value)
+        {
+            try
+            {
+                _mtx.WaitOne();
+                if (_map.TryGetValue(key, out Node node))
+                {
+                    node.Value = value;
+                    MoveToFront(node);
+                }
+                else
+                {
+                    if (_map.Count == _capacity)
```

Comment:
I understand that with the locking it should be impossible to have this condition, but it might be safer to make this a `>=` just in case.